feedback/views.py

```diff
@@ -483,20 +505,30 @@ def update_context_for_feedbacks(request, context, course=None, feedbacks=None,
                                            lambda c, t: (c.id, t.id))
 
     course_context_tags = ContextTag.objects.filter(course=course_id)
-    context_tag_groups: Dict[str, Dict[str, ContextTag]] = {} # {question_key: {response_value: ContextTag}}
+    context_tag_groups: dict[str, dict[str, ContextTag]] = {} # {question_key: {response_value: ContextTag}}
     for tag in course_context_tags:
         tag_group = context_tag_groups.setdefault(tag.question_key, {})
         tag_group[tag.response_value] = tag
 
+    # get background questionnaire urls
+    client = request.user.get_api_client(course.namespace)
+    course_has_bg_questionnaire = (
+        len(BackgroundCache.get_or_set_bg_questionnaires(course, client)) > 0 # pylint: disable=no-value-for-parameter
+    )
+
+    def student_has_no_bg_questionnaire(student: Student) -> bool:
+        resp = BackgroundCache.get_response(student, course) # pylint: disable=no-value-for-parameter
+        return (resp is not None) and (resp[0] is None)
```

Comment:
Can this be changed to `student_has_bg_questionnaire`, so that we don't have to use a double negative later on line 571? Currently the line 571 is a bit silly to read:

```python
'show_background': course_has_bg_questionnaire and not student_has_no_bg_questionnaire(conv.student),
```

Reply:
The problem is that this doesn't check if the student has a background questionnaire (as it a student does, but it hasn't been computed yet, it wouldn't appear in the cache). This checks if it has been checked and they student didn't have a background questionnaire (so it doesn't need to be computed and a background questionnaire doesn't need to be displayed). This basically means that the button will show up the first time for a student even if they don't have a background questionnaire, and then if the button is pressed (and the API request is made and we discover the student doesn't have a response to the bg questionnaire, it's saved in the cache so we don't show the button falsely in the future. I guess this could be renamed to "student_might_have_bg_questionnare" or then the entire functionality could be changed so for all students if they don't have a response saved, we actually check if they have a response (and based on that show the button or not), which would slow down the loading somewhat.

Reply:
If its difficult then it doesn't have to be changed. I was just thinking if it was easy, it would be a nice little change.

Reply:
It's not difficult, it just slows down the initial loading time. (For example, if there are about 100 students whose responses haven't been loaded, when testing locally, loading the page with the feedbacks takes about 10 seconds instead of 2.)  But otherwise it does make the UX probably a bit nicer (since the button doesn't appear if the student hasn't filled in the enrollment questionnaire) and indeed does make the code a bit more legible. So it would be a tradeoff. Hard to say which would be better.
I would suspect that decreasing the load time (and fetching the student responses on demand, even if it does mean sometimes in vain showing the "background" button if a student doesn't have a response) would be preferable.

Reply:
Perhaps keeping the functionality as it is and changing the function name to "student_may_have_bg_questionnaire" might be a good way of making the code a bit more legible.